### PR TITLE
Don't warn about #nullable in disabled code:

### PR DESCRIPTION
**Pull request by chsienki**
- Check if in an active code region before warning about #nullable directives in lower lang versions
 - Add test

jcouv updated:
Fixes https://github.com/dotnet/roslyn/issues/32172

### CODE REVIEW

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please

**Comment by jcouv** [comment]
Fixes https://github.com/dotnet/roslyn/issues/32172

File: src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs [hunk 1]
```diff
@@ -86641,5 +86641,32 @@ static void Main()
...
+            CreateCompilation(new[] { source }, parseOptions: TestOptions.Regular7_3)
+                .VerifyDiagnostics(
+                // (9,2): error CS8652: The feature 'nullable reference types' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // #nullable disable // 1
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "nullable").WithArguments("nullable reference types").WithLocation(9, 2)
+                );
+
+            CreateCompilation(new[] { source }, options: WithNonNullTypesTrue())
+                .VerifyDiagnostics();
+        }
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs]: Please also test that `#nullable enable` doesn't enable the feature when it's in disabled code

**Comment by chsienki** [comment]
Fixed for #pragma too, and added extra test to cover.

---
In reply to: [475104468](https://github.com/dotnet/roslyn/pull/34319#issuecomment-475104468) [](ancestors = 475104468)

**Pull request merged**